packages/aws-cdk-lib/aws-logs/lib/log-group.ts

```diff
@@ -491,7 +491,9 @@ export class LogGroup extends LogGroupBase {
 
   constructor(scope: Construct, id: string, props: LogGroupProps = {}) {
     super(scope, id, {
-      physicalName: props.logGroupName,
+      physicalName: props.logGroupName ?? Lazy.string({
+        produce: () => Names.uniqueResourceName(this, { maxLength: 512, allowedSpecialCharacters: '-_' }),
+      }),
```

Comment:
this shouldn't change existing log group names will it ?

Reply:
@kaizencc 
This is my concern as well.
With this approach, we'll probably trigger a replacement of existing log groups that don't have `logGroupName` specified (unless I'm missing something).
I don't see a workaround, and the previous solution with `ArnLike` / `:log-group:*` may be too loose.
Thoughts?

Reply:
Workaround as always is a feature flag, and I agree that the previous solution is probably too loose. Give me a bit to think this one through and see what we should od. I think the safest thing is a feature flag but will ask around to make sure.

Reply:
Alternative proposal:

If we add a [kms:ViaService](https://docs.aws.amazon.com/kms/latest/developerguide/conditions-kms.html#conditions-kms-via-service) condition to the key, is that not good enough? It will allow us to avoid naming the log group.

Reply:
@rix0rrr 
Isn't it the same as
```
conditions: {
 ArnLike: {
  'kms:EncryptionContext:aws:logs:arn': `arn:${this.stack.partition}:logs:${this.env.region}:${this.env.account}:log-group:*`,
 },
},
```
?
It looks to me like we restrict permissions to the service in both cases (original problem).

Reply:
@rix0rrr 
Isn't it the same as
```
conditions: {
 ArnLike: {
  'kms:EncryptionContext:aws:logs:arn': `arn:${this.stack.partition}:logs:${this.env.region}:${this.env.account}:log-group:*`,
 },
},
```
?
It looks to me like we restrict permissions to the service in both cases (original problem).